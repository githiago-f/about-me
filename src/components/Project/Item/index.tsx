import React from 'react';
import { Project } from 'entities';

export const Item = ({name, description}: Project) => (
  <div data-testid="item" className="w-100">
    <div className="bg-white w-100 p-3 rounded-lg">
      <h3 translate="no">{name}</h3>
      <p className="text-gray-500">{description}</p>
    </div>
  </div>
);
