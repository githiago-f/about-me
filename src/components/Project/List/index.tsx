import React from 'react';
import { Item } from '../Item';

export const List = () => (
  <div className="min-h-screen bg-gray-100 p-3 gap-2 grid grid-cols-1 md:grid-cols-3 sm:grid-cols-2">
    <Item name="" description=""/>
    <Item name="" description=""/>
    <Item name="" description=""/>
    <Item name="" description=""/>
    <Item name="" description=""/>
    <Item name="" description=""/>
  </div>
);
