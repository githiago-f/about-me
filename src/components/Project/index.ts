import { Item } from './Item';
import { List } from './List';

const Project = {
  Item, List
};

export default Project;
