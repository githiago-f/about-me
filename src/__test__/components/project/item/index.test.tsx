import React from 'react';
import Project from 'components/Project';
import { render, screen } from '@testing-library/react';

describe('# Project Item', () => {
  beforeEach(() => {
    render(
      <Project.Item 
        description='Descrição do item que foi criado'
        name="Nome do projeto"
      />
    );
  });

  it('should render properly', async () => {
    const item = await screen.findByTestId('item');
    expect(item).toMatchSnapshot();
  });
});
