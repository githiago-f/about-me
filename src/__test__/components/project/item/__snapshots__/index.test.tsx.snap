// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Project Item should render properly 1`] = `
<div
  class="w-100"
  data-testid="item"
>
  <div
    class="bg-white w-100 p-3 rounded-lg"
  >
    <h3
      translate="no"
    >
      Nome do projeto
    </h3>
    <p
      class="text-gray-500"
    >
      Descrição do item que foi criado
    </p>
  </div>
</div>
`;
